pkg/config/cmd/cmd.go

```diff
@@ -4,23 +4,74 @@ import (
 	"fmt"
 	"io"
 
+	"github.com/spf13/pflag"
+
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/labels"
 	"k8s.io/kubernetes/pkg/runtime"
 )
 
+type Runner interface {
+	Run(list *kapi.List, namespace string) []error
+}
+
 // AfterFunc takes an info and an error, and returns true if processing should stop.
 type AfterFunc func(*resource.Info, error) bool
 
+// OpFunc takes the provided info and attempts to perform the operation
+type OpFunc func(info *resource.Info, namespace string, obj runtime.Object) (runtime.Object, error)
+
 // Bulk provides helpers for iterating over a list of items
 type Bulk struct {
-	Mapper            meta.RESTMapper
-	Typer             runtime.ObjectTyper
-	RESTClientFactory func(mapping *meta.RESTMapping) (resource.RESTClient, error)
-	After             AfterFunc
-	Retry             func(info *resource.Info, err error) runtime.Object
+	Mapper resource.Mapper
+
+	Op    OpFunc
+	After AfterFunc
+	Retry func(info *resource.Info, err error) runtime.Object
+}
+
+// Create attempts to create each item generically, gathering all errors in the
+// event a failure occurs. The contents of list will be updated to include the
+// version from the server.
+func (b *Bulk) Run(list *kapi.List, namespace string) []error {
+	after := b.After
+	if after == nil {
+		after = func(*resource.Info, error) bool { return false }
+	}
+
+	errs := []error{}
+	for i, item := range list.Items {
+		info, err := b.Mapper.InfoForObject(item, nil)
+		if err != nil {
+			errs = append(errs, err)
+			if after(info, err) {
+				break
+			}
+			continue
+		}
+		obj, err := b.Op(info, namespace, item)
+		if err != nil && b.Retry != nil {
+			if obj := b.Retry(info, err); obj != nil {
```

Comment:
That's pre-existing - I think we can came back to it later and change it.

On Mon, Apr 18, 2016 at 8:28 AM, David Eads <notifications@github.com>
wrote:

> In pkg/config/cmd/cmd.go
> <https://github.com/openshift/origin/pull/8539#discussion_r60048777>:
>
> > +		after = func(*resource.Info, error) bool { return false }
> > +	}
> > +
> > +	errs := []error{}
> > +	for i, item := range list.Items {
> > +		info, err := b.Mapper.InfoForObject(item, nil)
> > +		if err != nil {
> > +			errs = append(errs, err)
> > +			if after(info, err) {
> > +				break
> > +			}
> > +			continue
> > +		}
> > +		obj, err := b.Op(info, namespace, item)
> > +		if err != nil && b.Retry != nil {
> > +			if obj := b.Retry(info, err); obj != nil {
>
> Since retry seems a little strange. How about a struct value saying how
> many times it can vote for yes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8539/files/e44ca5593b4d8049f941c4573a5a245248f6481a#r60048777>
>